### ISSUE TITLE
StaticWebAssets manifest is rewritten with no data if a project depends on a Blazor app

### ISSUE THREAD

**Issue by bwateratmsft** [dotnet/aspnetcore#21828]
### Describe the bug
Originally from here: https://developercommunity.visualstudio.com/content/problem/1023875/staticwebassetsxml-file-content-missing-for-class.html

In the Visual Studio docker-compose tooling, a docker-compose.dcproj project is created that orchestrates all the necessary stuff to build and run in containers. This project is treated as the startup project and depends on the actual web apps, e.g. a Blazor server web app.

When an App B depends on Blazor Server Web App A, A will be built (creating the manifest with the right info), then B will be built (wiping out the manifest).

### To Reproduce
1. Create a Razor class library
1. Create a Blazor server web app, and reference the above library, so that something will be generated in static web assets
1. Create another project, e.g. a console app, which references the Blazor app (this is contrived, obviously a console app would not depend on a website, but this is a simpler example than doing it with the compose tooling).
1. Build the Blazor app in Visual Studio. Note that in `obj\Debug\netcoreapp3.1\staticwebassets\MyApp.StaticWebAssets.xml`, an entry is created referencing the Razor library.
1. Build the console app in Visual Studio. Note that now the above manifest will have no `<ContentRoot>` entries.

### Further technical details
- ASP.NET Core 3.1
- VS 2019 version 16.5.5

### Additional notes
The reason this is breaking for Compose is that, in order to make static web assets work, it is necessary for the compose tooling to alter the manifest. It does this by volume mapping relevant directories, and then changing the host-side paths in the manifest to container-side paths (and changing slash direction for Linux containers). Without the manifest and this transformation of it, it is impossible to use static web assets in containers.

The user can fall back to debugging in "Regular" mode for compose--which means fully building and publishing the site--but this is dramatically slower.

**Comment by javiercn** [comment]
@bwateratmsft thanks for filing the issue.

We'll triage this today and discuss it further within the team

**Comment by jrcpereira** [comment]
Any way we can circumvent this?

**Comment by LeonSebastianCoimbra** [comment]
I think that this post on S.O. is very useful:

https://stackoverflow.com/questions/59228315/debug-blazor-wasm-using-visual-studio-container-tools

**Comment by john-cullen** [comment]
Is there a workaround for this? The linked StackOverflow answer does not work for me.

**Comment by javiercn** [comment]
This was fixed in 5.0. I just tried it with docker-compose which was the original source of this issue.

**Comment by john-cullen** [comment]
@javiercn Does something need to be done to get this to work? I've installed RC2 and adding a blazor wasm to my MVC app still does not work even for the simple test case of replacing / with the page generated by the template.

I have performed the tasks listed at https://docs.microsoft.com/en-us/aspnet/core/migration/31-to-50?view=aspnetcore-3.1&tabs=visual-studio

Do I need to update part of the docker-compose project or yaml files?

**Comment by javiercn** [comment]
@john-cullen I'm not sure about any changes in the docker-compose or the yaml files.

I created a new project, added docker-compose support and hit run on it, and I saw the manifest had the right data and everything was working as expected.

**Comment by john-cullen** [comment]
@javiercn you were correct, I had a mistake in my startup.cs and now it works on dotnet 5 RC 2. However, I can't get WASM debugging to work. Is this known not to work or have I got another mistake?

(`UseWebAssemblyDebugging();` is called, doing `shift-alt-D` opens a new tab with the debugging URL but it returns a 404)

**Comment by javiercn** [comment]
> @javiercn you were correct, I had a mistake in my startup.cs and now it works on dotnet 5 RC 2. However, I can't get WASM debugging to work. Is this known not to work or have I got another mistake?
> 
> (`UseWebAssemblyDebugging();` is called, doing `shift-alt-D` opens a new tab with the debugging URL but it returns a 404)

@john-cullen I would file a separate issue for that so that we don't loose track of it, but I do believe WASM debugging is not something we currently support within containers. @captainsafia probably knows more.

**Comment by captainsafia** [comment]
@john-cullen Are you trying to debug a Blazor app inside a Docker container? If so, this experience isn't currently supported. I can show you how to work around this locally to unblock yourself.

If you're trying to debug a Blazor app that's not inside a Docker container, can you open a new issue with the relevant details?

**Comment by john-cullen** [comment]
@captainsafia It is inside a Docker container. 

Is there an issue I can follow that tracks this? (assuming it is something you plan to support in future?), and if you could point me at a work-around that would be helpful.
